File: csharp/main/InterfacePropertyDef.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public class InterfacePropertyDef : PropertyDef<InterfaceDef>
	{
		public InterfacePropertyDef([NotNull] InterfaceDef interfaceDef, [NotNull] String name, [NotNull] Type type, Boolean isOverride, Boolean isReadable, Boolean isWritable, [CanBeNull, ItemNotNull] List<String> comments = null)
			: base(interfaceDef, name, type, comments)
		{
			IsReadable = isReadable;
			IsWritable = isWritable;
			IsOverride = isOverride;

			FullName = $"{interfaceDef.FullName}.{Name}";

			interfaceDef.Properties.Add(this);
		}

		/// <summary>
		/// Indicates whether this property overrides some property from the base class.
		/// </summary>
		public Boolean IsOverride { get; }

		public Boolean IsReadable { get; }

		public Boolean IsWritable { get; }

		public override String FullName { get; }
	}
}

File: java/src/main/java/deltix/luminary/InterfacePropertyDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class InterfacePropertyDef extends PropertyDef<InterfaceDef> {
    private final boolean isOverride;
    private final boolean isReadable;
    private final boolean isWritable;

    public InterfacePropertyDef(@NotNull InterfaceDef interfaceDef, @NotNull String name, @NotNull Type type,
                                boolean isOverride, boolean isReadable, boolean isWritable,
                                @Nullable List<String> comments) {
        super(interfaceDef, name, type, comments);
        this.isOverride = isOverride;
        this.isReadable = isReadable;
        this.isWritable = isWritable;

        interfaceDef.getProperties().add(this);
    }

    public InterfacePropertyDef(@NotNull InterfaceDef interfaceDef, @NotNull String name, @NotNull Type type,
                                boolean isOverride, boolean isReadable, boolean isWritable) {
        this(interfaceDef, name, type, isOverride, isReadable, isWritable, null);
    }

    public boolean isOverride() {
        return isOverride;
    }

    public boolean isReadable() {
        return isReadable;
    }

    public boolean isWritable() {
        return isWritable;
    }
}
